src/common.rs

```diff
@@ -212,52 +213,38 @@ pub async fn poll_for_p2p(
     ans_vec
 }
 
-/*
-pub fn check_sig(
-    r: &Scalar,
-    s: &Scalar,
-    msg: &BigInt,
-    pk: &Point,
-) {
-
-    let raw_msg = BigInt::to_bytes(msg);
-    let mut msg: Vec<u8> = Vec::new(); // padding
-    msg.extend(vec![0u8; 32 - raw_msg.len()]);
-    msg.extend(raw_msg.iter());
-
-    let msg = Message::from_slice(msg.as_slice()).unwrap();
-    let mut raw_pk = pk.to_bytes(false).to_vec();
-    if raw_pk.len() == 64 {
-        raw_pk.insert(0, 4u8);
-    }
-    let pk = PublicKey::from_slice(&raw_pk).unwrap();
-
-    let mut compact: Vec<u8> = Vec::new();
-    let bytes_r = &r.to_bytes().to_vec();
-    compact.extend(vec![0u8; 32 - bytes_r.len()]);
-    compact.extend(bytes_r.iter());
+pub fn check_sig(r: &Scalar, s: &Scalar, msg: &BigInt, pk: &Point) {
+    use secp256k1::*;
+    let r_vec = BigInt::to_vec(&r.to_big_int());
+    let s_vec = BigInt::to_vec(&s.to_big_int());
 
-    let bytes_s = &s.to_bytes().to_vec();
-    compact.extend(vec![0u8; 32 - bytes_s.len()]);
-    compact.extend(bytes_s.iter());
-
-    let secp_sig = Signature::from_compact(compact.as_slice()).unwrap();
+    let mut signature_a = [0u8; 64];
+    for i in 0..32 {
+        signature_a[i] = r_vec[i];
+    }
+    for i in 0..32 {
+        signature_a[i + 32] = s_vec[i];
+    }
+    let signature = Signature::parse(&signature_a);
 
-    let is_correct = SECP256K1.verify(&msg, &secp_sig, &pk).is_ok();
+    let message = Message::parse(&BigInt::to_vec(msg).try_into().unwrap());
 
-    println!("public key: {:}", pk);
-    println!("address: {:}", public_key_address(&pk));
+    let pk_vec = BigInt::to_vec(&pk.bytes_compressed_to_big_int());
 
+    let mut pubkey_a = [0u8; 65];
+    for i in 0..65 {
+        pubkey_a[i] = pk_vec[i];
+    }
+    let pubkey = PublicKey::parse(&pubkey_a).unwrap();
 
-    assert!(is_correct);
+    assert!(verify(&message, &signature, &pubkey));
```

Comment:
don't assert here, return the error.

Reply:
where is the verify from?

Reply:
check out the example here:  https://github.com/paritytech/libsecp256k1/blob/v0.3.5/tests/verify.rs#L57

Reply:
OK.